src/apis.md

```diff
@@ -0,0 +1,165 @@
+
+There's a reasonable chance that your API can be used in parallel threads
+by virtue of `Send` and `Sync` being automatically derived. But - you should
+think through the usage model for your API clients and ensure that's true.
+
+```rust
+use std::cell::RefCell;
+use std::collections::VecDeque;
+use std::sync::Mutex;
+use std::thread;
+
+// Absolutely no effort required to pizza, cooking or eating procedures
+// to be usable in a threaded context
+mod pizza_api {
+
+    use std::thread;
+    use std::time::Duration;
+
+    pub struct Pizza {
+        // automatically 'Send'
+        _anchovies: u32,
+        _pepperoni: u32,
+    }
+
+    pub fn make_pizza() -> Pizza {
+        println!("cooking...");
+        thread::sleep(Duration::from_millis(10));
+        Pizza {
+            _anchovies: 0, // yuck
+            _pepperoni: 32,
+        }
+    }
+
+    pub fn eat_pizza(_pizza: Pizza) {
+        println!("yum")
+    }
+}
+
+fn main() {
+    let pizza_queue = Mutex::new(RefCell::new(VecDeque::new()));
+    thread::scope(|s| {
+        s.spawn(|| {
+            let mut pizzas_eaten = 0;
+            while pizzas_eaten < 100 {
+                if let Some(pizza) = pizza_queue.lock().unwrap().borrow_mut().pop_front() {
+                    pizza_api::eat_pizza(pizza);
+                    pizzas_eaten += 1;
+                }
+            }
+        });
+        s.spawn(|| {
+            for _ in 0..100 {
+                let pizza = pizza_api::make_pizza();
+                pizza_queue.lock().unwrap().borrow_mut().push_back(pizza);
+            }
+        });
+    });
+}
+
+
+```
+
+## What should I `Derive` to make my code optimally usable?
+
+The [official guidelines say to be eager](https://rust-lang.github.io/api-guidelines/interoperability.html#types-eagerly-implement-common-traits-c-common-traits).
+
+But don't overpromise:
+
+> Equality can suddenly become expensive later - don’t make types comparable
+> unless you intend people to be able to do that. Allowing people to pattern
+> match on enums is usually better - MY.
+
+Note that [`syn` is a rare case](https://docs.rs/syn/latest/syn/) where it
+has so many types, and is so extensively depended upon by the rest of the Rust
+ecosystem, that it avoids deriving the standard traits unless explicitly
+commanded to do so via cargo feature. This is an unusual pattern and should
+not normally be followed.
+
+## How should I think about API design, differently from C++?
+
+> make the most of the fact that everything is immutable by default. It should
+> be trivially observable that most things in the program do not change - the
+> language encourages everything to be immutable - things which are mutable
+> should stick out. - AF.
+
+> think about things which should take self and return self - AF.
+
+Refactoring is less expensive in Rust than C++ due to compiler safeguards, but
+_restructuring_ is expensive in any language. Think about "one way doors"
+and "two way doors" in the design space.
```

Comment:
I'm not sure what the doors metaphor means. I'm also not sure what the difference between refactoring and restructuring is.